torch/csrc/api/include/torch/nn/module.h

```diff
@@ -148,4 +148,11 @@ std::shared_ptr<typename std::decay<Module>::type> make(Module&& module) {
   ptr->reset_parameters();
   return ptr;
 }
+
+/// A type trait whose `::value` member is true if `M` derives from `Module`.
+template <typename M>
+using is_module = std::is_base_of<Module, typename std::decay<M>::type>;
+
+template <typename M>
+using enable_if_module = typename std::enable_if<is_module<M>::value>::type;
```

Comment:
Without the `_t` prefix, convention is to not go into the `::type` member, i.e.

    template <typename M>
    using enable_if_module = std::enable_if<is_module<M>::value>;
   
    template <typename M>
    using enable_if_module_t = typename enable_if_module<M>::type;